### DISCUSSION TITLE
[rtk-query] what is the purpose of unsubscribe when dispatching initiate manually

### DISCUSSION THREAD

**Discussion by cjhdom** [reduxjs/redux-toolkit#3012] [Q&A] [answered]
https://redux-toolkit.js.org/rtk-query/api/created-api/endpoints#initiate

in the doc above states

>  Also, you have to manually unsubscribe once your component unmounts.

I am struggling to understand what exactly it does and when it's and it's not needed.
example code shown in the docs doesn't seem to call unsubscribe at all

Looking at the source code, it seems like resetting cached data when unsubscribe (deprecated, to use reset) is called
only has meaning when called for a mutation not a query. is that correct?

to summarize
is calling unsubscribe only necessary when calling mutation with initialte?

Thank you

**Comment by markerikson** [comment, marked as answer]
@cjhdom : 

RTK Query uses a form of reference counting to know how many "subscriptions" still care about a given cache entry.  Normally, one component+query hook call === one subscription.  So, if you had 10 components that all all `useGetSomeDataQuery("abc")`, that's 10 active subscriptions to that cache entry.

When the count of active subscriptions goes to 0, RTKQ sets a timer, and if the subscription count is still 0 when the timer expires, RTKQ will remove that cache entry:

- https://redux-toolkit.js.org/rtk-query/usage/cache-behavior#default-cache-behavior

This works because the query hooks automatically unsubscribe in `useEffect` cleanup functions, so when a component unmounts it tells RTKQ that it no longer cares about the data.

If you're manually dispatching `initiate()` yourself, then it's up to you to decide if and when to do that unsubscribing in your own code.

You can see an example of unsubscribing manually here:

- https://redux-toolkit.js.org/rtk-query/usage/examples#react-class-components

**Reply by cjhdom** [reply]
thanks a lot! 
I had tried to see what's going on with some test code but seems like I missed the 60 seconds part
(although as mentioned I still couldn't find `unsubscribe` nor `componentWillUnmount` in the example code)
